app/gui2/src/components/GraphEditor.vue

```diff
@@ -441,8 +453,50 @@ function openComponentBrowser(usage: Usage, position: Vec2) {
 function hideComponentBrowser() {
   graphStore.editedNodeInfo = undefined
   componentBrowserVisible.value = false
+  displayedDocs.value = null
 }
 
+const rightDockDisplayedTab = computed({
+  get() {
+    if (componentBrowserVisible.value) {
+      if (userSettings.value.showHelpForCB) {
+        return 'help'
+      } else {
+        return showRightDock.value ? rightDockTab.value : rightDockTabForCB.value
+      }
+    } else {
+      return rightDockTab.value
+    }
+  },
+  set(tab) {
+    if (componentBrowserVisible.value) {
+      rightDockTabForCB.value = tab
+      userSettings.value.showHelpForCB = tab === 'help'
+      if (showRightDock.value) rightDockTab.value = tab
+    } else {
+      rightDockTab.value = tab
+    }
+  },
+})
+const rightDockVisible = computed({
+  get() {
+    if (componentBrowserVisible.value) {
+      return userSettings.value.showHelpForCB || rightDockVisibleForCB.value || showRightDock.value
+    } else {
+      return showRightDock.value
+    }
+  },
+  set(vis) {
+    if (componentBrowserVisible.value) {
+      rightDockVisibleForCB.value = vis
+      userSettings.value.showHelpForCB = vis
+      if (!vis) showRightDock.value = false
+    } else {
+      showRightDock.value = vis
+    }
+  },
+})
```

Comment:
I think the application logic could be made easier to read here by creating a helper like `useSelectRef<T>(select: ToValue<Ref<T>>): Ref<T>`

Reply:
I’m not quite following the idea :( Let’s discuss is on a call?